### PR TITLE
create new mesh layer

### PR DESCRIPTION
**Pull request by vcloarec**
In the scope of mesh layer editing (https://github.com/qgis/QGIS-Enhancement-Proposals/issues/228), the aim of this PR is to allow the user to create new frame of mesh layer.
From the dialog, the user can create:
* an empty mesh layer
* a frame based on an existing mesh in the project
* a frame based on a existing mesh from a file

To work fine, this new functionality needs few changes in MDAL that are coming...

createNewMesh_1

createNewMesh_2

createNewMesh_3

### CODE REVIEW

File: src/ui/mesh/qgsnewmeshlayerdialogbase.ui [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+   </rect>
+  </property>
+  <property name="windowTitle">
+   <string>New Mesh Layer</string>
+  </property>
+  <layout class="QFormLayout" name="formLayout">
+   <item row="0" column="0">
+    <widget class="QLabel" name="label_3">
+     <property name="text">
+      <string>Layer Name</string>
```
Comment (nyalldawson) [inline src/ui/mesh/qgsnewmeshlayerdialogbase.ui:20]:
```suggestion
      <string>Layer name</string>
```

File: src/ui/mesh/qgsnewmeshlayerdialogbase.ui [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+     </property>
+    </widget>
+   </item>
+   <item row="0" column="1">
+    <widget class="QLineEdit" name="mLayerNameLineEdit">
+     <property name="text">
+      <string>New Mesh Layer</string>
```
Comment (nyalldawson) [inline src/ui/mesh/qgsnewmeshlayerdialogbase.ui:27]:
```suggestion
      <string>New mesh layer</string>
```

File: src/ui/mesh/qgsnewmeshlayerdialogbase.ui [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+     </property>
+    </widget>
+   </item>
+   <item row="1" column="1">
+    <widget class="QgsFileWidget" name="mFileWidget" native="true"/>
+   </item>
+   <item row="2" column="0">
+    <widget class="QLabel" name="label">
+     <property name="text">
+      <string>File Format</string>
```
Comment (nyalldawson) [inline src/ui/mesh/qgsnewmeshlayerdialogbase.ui:44]:
```suggestion
      <string>File format</string>
```

File: src/ui/mesh/qgsnewmeshlayerdialogbase.ui [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+   <item row="2" column="1">
+    <widget class="QComboBox" name="mFormatComboBox"/>
+   </item>
+   <item row="3" column="1">
+    <widget class="QgsProjectionSelectionWidget" name="mProjectionSelectionWidget" native="true"/>
+   </item>
+   <item row="4" column="0" colspan="2">
+    <widget class="QGroupBox" name="groupBox">
+     <property name="title">
+      <string>Frame of the New Mesh</string>
```
Comment (nyalldawson) [inline src/ui/mesh/qgsnewmeshlayerdialogbase.ui:57]: I must admit I don't understand this string at all -- what does it mean?

Comment (vcloarec) [reply src/ui/mesh/qgsnewmeshlayerdialogbase.ui:57]: Frame is the faces/vertices of the mesh, without dataset. Here it is on which frame the new mesh will be based.

I confess I have some difficulties to find a good string ... Maybe "Base of the new mesh frame"

Comment (nyalldawson) [reply src/ui/mesh/qgsnewmeshlayerdialogbase.ui:57]: What exactly does this setting do though? Does it copy the data from the selected layer into the new mesh?

Comment (vcloarec) [reply src/ui/mesh/qgsnewmeshlayerdialogbase.ui:57]: yes, but only the faces/vertices, not dataset. If user want to create and edit a new mesh, he can use an existing one.

Comment (nyalldawson) [reply src/ui/mesh/qgsnewmeshlayerdialogbase.ui:57]: Hmm... Maybe "Initialize Mesh Using"?

**Comment by PeterPetrik** [comment]
@vcloarec why this has label "not for merge?"

**Comment by vcloarec** [comment]
@PeterPetrik , because waiting of MDAL update to finalize the PR and to be sure the QGIS community doesn't see objection for this new feature.
Now the PR is finalized.

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**